kernel/src/process/posix_thread/exit.rs

```diff
@@ -26,9 +26,7 @@ pub fn do_exit(thread: &Thread, posix_thread: &PosixThread, term_status: TermSta
     if *clear_ctid != 0 {
         futex_wake(*clear_ctid, 1, None)?;
         // FIXME: the correct write length?
-        get_current_userspace!()
-            .write_val(*clear_ctid, &0u32)
-            .unwrap();
+        get_current_userspace!().write_val(*clear_ctid, &0u32)?;
```

Comment:
I think we should keep going on with the exit process. The status of the thread has been set to exited. There is no turning back. In addition, I don't think the exit system call should ever fail.

Reply:
The updated code ignores the error and keeps going the exit process.

To prevent the exit system call from failure, maybe Asterinas could add some protection to these critical memory addresses?

Reply:
No, I don't think so. As a kernel, Asterinas cannot prevent user code from messing up with its own memory. It is the user program's responsibility to follow the contract agreed upon by both the user program and the kernel. Asterinas only needs to make sure that a process can only mess up with the resources that it is allowed to access, not those that belong to other processes.